### DISCUSSION TITLE
Password reset redirect in "WPF" or .net  MAUI

### DISCUSSION THREAD

**Discussion by makgahlela** [supabase-community/supabase-csharp#111] [Q&A] [answered]
how to get the app to detect type=recovery in the fragment of the url below  and display a password reset form to the user using supabase c#  in "WPF" or .net  MAUI ( after the user have clicked the 'reset password' in the email) 
https://www.domainName.com/#access_token=........&expires_at=1696000393&expires_in=9600&refresh_token=S2158752dhjjjj77niUA&token_type=bearer&type=recovery
![ResettingPassword](https://github.com/supabase-community/supabase-csharp/assets/55931402/825afc89-bd08-4260-9b06-2c138ca199ec)

**Comment by acupofjose** [comment, marked as answer]
Good catch - this is a method that requires PKCE support. The gotrue team intentionally set the redirect urls to use a `#` instead of a `?` so that the parameters passed back can only be read on the client side (so C# has a _real_ hard time playing nice with these). A PKCE flow gets around this, but that hasn't been implemented on our client yet (didn't know about this functionality!).

That is now being tracked here: https://github.com/supabase-community/supabase-csharp/issues/112

**Comment by makgahlela** [comment]
Thank you for the update and for tracking the issue (#112). It's good to
know that PKCE support is being considered to address the use of "#" in the
redirect URLs for enhanced security. I look forward to seeing the
implementation of PKCE in the future. If there's any way I can assist or
contribute to this effort, please don't hesitate to reach out.

On Sun, Oct 1, 2023 at 4:30 PM Joseph Schultz ***@***.***>
wrote:

> Good catch - this is a method that requires PKCE support. The gotrue team
> intentionally set the redirect urls to use a # instead of a ? so that the
> parameters passed back can only be read on the client side. A PKCE flow
> gets around this, but that hasn't been implemented on our client yet
> (didn't know about this functionality!).
>
> That is now being tracked here: #112
> <https://github.com/supabase-community/supabase-csharp/issues/112>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/supabase-community/supabase-csharp/discussions/111#discussioncomment-7157858>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKXECSIGRHKEHUZPVTAQNDX5F5ADANCNFSM6AAAAAA5NWJABE>
> .
> You are receiving this because you authored the thread.Message ID:
> <supabase-community/supabase-csharp/repo-discussions/111/comments/7157858@
> github.com>
>


-- 
MR. MELVIN MAGORO

**Reply by acupofjose** [reply]
It should be available for you in 0.13.4!

You can find an example of this in the "wild" using github login (essentially the same process)
https://github.com/acupofjose/supasharp-todo/tree/master/SupasharpTodo.BlazorWASM/Shared
